Source/UZKFileInfo.m

```diff
@@ -65,6 +97,24 @@ - (NSDate *)timestamp {
 
 
 
+#pragma mark - Private Class Methods
+
+
++ (BOOL)itemIsDirectory:(unz_file_info64 *)fileInfo {
+    UZKZipOS zipOSClassMapping = fileInfo->version >> 8;
+    if (zipOSClassMapping == UZKZipOSMSDOS || zipOSClassMapping == UZKZipOSWindowsNT) {
+        return 0x01 == (fileInfo->external_fa >> 4) && ![UZKFileInfo itemIsSymbolicLink:fileInfo];
+    }
+    
+    return S_IFDIR == (S_IFMT & fileInfo->external_fa >> 16) && ![UZKFileInfo itemIsSymbolicLink:fileInfo];
```

Comment:
* Why use the `S_IFDIR` and `S_IFMT` constants?  The names don't make it very clear what we're checking against. `S_IFDIR` is probably "is dir" and `S_IFLNK` (below) is probably "is link", but what's `S_IFMT`, and how did you even come across these constants? I'm curious
* Like I suggested above, let's encapsulate the `S_IFMT & fileInfo->external_fa >> 16` expression into a class method explaining the field it's retrieving, and comparing to constants - unless it's boolean, which it looks like it may be. Then just place the boolean call inline where the expression is
* Also as mentioned above, either we write a test case that hits the ``YES && NO` condition here, or we drop the `&&` clause

Reply:
About using S_IFDIR, S_IFLNK and S_IFMT constants is to get the type of zip entry more accurately, get Unix Filemodel by (external_fa >> 16).
You can get more information here:
Https://unix.stackexchange.com/questions/14705/the-zip-formats-external-file-attribute/14727#14727

About using 0x01 == (fileInfo->external_fa >> 4)
0x01 represents a folder in WinZip, the specific value is from Google chromium project line 4000
zip has an 'attribute' 32bit value.
Its lower half is windows stuff, its upper half is standard unix attr.